mm2src/lp_bot/simple_market_maker.rs

```diff
@@ -0,0 +1,266 @@
+    fn from(error: String) -> Self { PriceServiceRequestError::HttpProcessError(error) }
+}
+
+impl From<std::str::Utf8Error> for PriceServiceRequestError {
+    fn from(error: Utf8Error) -> Self { PriceServiceRequestError::HttpProcessError(error.to_string()) }
+}
+
+impl HttpStatusCode for StartSimpleMakerBotError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            StartSimpleMakerBotError::AlreadyStarted | StartSimpleMakerBotError::InvalidBotConfiguration => {
+                StatusCode::BAD_REQUEST
+            },
+            StartSimpleMakerBotError::Transport(_) | StartSimpleMakerBotError::InternalError(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+impl HttpStatusCode for StopSimpleMakerBotError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            // maybe bad request is not adapted for the first errors.
+            StopSimpleMakerBotError::AlreadyStopped | StopSimpleMakerBotError::AlreadyStopping => {
+                StatusCode::BAD_REQUEST
+            },
+            StopSimpleMakerBotError::Transport(_) | StopSimpleMakerBotError::InternalError(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+pub async fn tear_down_bot(ctx: MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+    {
+        let mut trading_bot_cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
+        // todo: check if clear is adapted, if i understand its keep the memory allocated for later usage.
+        trading_bot_cfg.clear();
+    }
+    // todo: cancel all pending orders
+}
+
+async fn update_single_order(_cfg: SimpleCoinMarketMakerCfg, uuid: Uuid, _order: MakerOrder, key_trade_pair: String) {
+    info!("need to update order: {} of {}", uuid, key_trade_pair)
+}
+
+async fn process_bot_logic(ctx: &MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
+    // note: Copy the cfg here will not be expensive, and this will be thread safe.
+    let cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await.clone();
+
+    let mut memoization_pair_registry: HashSet<String> = HashSet::new();
+    let maker_orders = retrieve_my_maker_orders(ctx).await;
+
+    info!("np_orders: {}", maker_orders.len());
+    for (key, value) in maker_orders.into_iter() {
+        let key_trade_pair = value.base.clone() + "/" + value.rel.clone().as_str();
+        match cfg.get(&key_trade_pair) {
+            Some(coin_cfg) => {
+                update_single_order(coin_cfg.clone(), key, value.clone(), key_trade_pair.clone()).await;
+                memoization_pair_registry.insert(key_trade_pair.clone());
+            },
+            _ => continue,
+        }
+        println!("{}", key);
+    }
+}
+
+pub async fn lp_bot_loop(ctx: MmArc) {
+    info!("lp_bot_loop successfully started");
+    loop {
```

Comment:
It looks like a good chance to start more extensive usage of the `State machine` pattern implemented by @sergeyboyko0791 :slightly_smiling_face: Please check https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1/mm2src/common/patterns/state_machine.rs. We plan to refactor some code parts to it in the future too.